src/modules/python/pbs_hooks/PBS_power.PY

```diff
@@ -0,0 +1,191 @@
+"""
+/* 
+#  Copyright (C) 1994-2017 Altair Engineering, Inc.
+#  For more information, contact Altair at www.altair.com.
+#   
+#  This file is part of the PBS Professional ("PBS Pro") software.
+#  
+#  Open Source License Information:
+#   
+#  PBS Pro is free software. You can redistribute it and/or modify it under the
+#  terms of the GNU Affero General Public License as published by the Free 
+#  Software Foundation, either version 3 of the License, or (at your option) any 
+#  later version.
+#   
+#  PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY 
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+#  PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details.
+#   
+#  You should have received a copy of the GNU Affero General Public License along 
+#  with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   
+#  Commercial License Information: 
+#  
+#  The PBS Pro software is licensed under the terms of the GNU Affero General 
+#  Public License agreement ("AGPL"), except where a separate commercial license 
+#  agreement for PBS Pro version 14 or later has been executed in writing with Altair.
+#   
+#  Altair’s dual-license business model allows companies, individuals, and 
+#  organizations to create proprietary derivative works of PBS Pro and distribute 
+#  them - whether embedded or bundled with other software - under a commercial 
+#  license agreement.
+#  
+#  Use of Altair’s trademarks, including but not limited to "PBS™", 
+#  "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's 
+#  trademark licensing policies.
+ *
+ */
+"""
+
+__doc__ = """
+$Id$
+
+This hook will activate and deactivate a power profile contained
+in the 'eoe' value for a job.
+"""
+
+import pbs
+import os
+from pbs.v1._pmi_utils import _get_vnodes, _svr_vnode
+
+
+def init_power(event):
+    # Get a Power structure and do the connect.  Reject on failure.
+    try:
+        confvar = "PBS_PMINAME"
+        name = None
+        if confvar in os.environ:
+            name = os.environ[confvar]
+        power = pbs.Power(name)
+        power.connect()
+    except Exception, e:
+        event.reject(str(e))
+    return power
+
+
+def vnodes_enabled(job):
+    # see if power operations are allowed on all job vnodes
+    for vn in _get_vnodes(job):
+        if not _svr_vnode(vn).power_provisioning:
+            pbs.logjobmsg(job.id,
```

Comment:
why do we want to log it in the job context for all the existing jobs on the vnode? Can't we just log it once ?